src/api/transfer.js

```diff
@@ -13,13 +13,13 @@ const validateInputs = validateWithJoi(schema)('INVALID_METHOD_ARGUMENT')({
   context: `transfer`
 })
 
-module.exports = async (dvf, data, starkPrivateKey) => {
+module.exports = async (dvf, data, starkPrivateKey, nonce, signature) => {
```

Comment:
@adrian-gierakowski Can you confirm if there is an issue with passing nonce and signature down especially for users on `usingTradingKey` flow

Reply:
I don't think there should be a problem with this, however going forward it would be ideal if we implemented an `authProvider`, which would abstract creation of signature for request authentication in similar way as `dvfStarkProvider` does for signing stark transactions. This would remove the need for passing nonce/signature to each method which requires authentication